scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,264 @@
+	exchange: 'erc20',
+	twinTokenSymbol: 'ckSepolia${tokenDetails.symbol}'
+};
+`;
+
+	await fs.writeFile(newFilePath, newContent);
+	console.log(`Created ${newFilePath}`);
+
+	return { newFileName, mainnetToken, testnetToken };
+};
+
+const updateTokensErc20Env = async (newFileName, mainnetToken, testnetToken) => {
+	const filePath = path.join(SRC_PATH, 'tokens.erc20.env.ts');
+	let content = await fs.readFile(filePath, 'utf8');
+
+	const mainnetRegex = /const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token\[] = \[([\s\S]*?)];/;
+	const sepoliaRegex = /const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token\[] = \[([\s\S]*?)];/;
+
+	const mainnetMatch = content.match(mainnetRegex);
+	const sepoliaMatch = content.match(sepoliaRegex);
+
+	if (mainnetMatch) {
+		const tokensList = mainnetMatch[1].trim();
+		const updatedTokensList = tokensList ? `${tokensList}, ${mainnetToken}` : mainnetToken;
+		content = content.replace(
+			mainnetRegex,
+			`const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token[] = [${updatedTokensList}];`
+		);
+	}
+
+	if (sepoliaMatch) {
+		const tokensList = sepoliaMatch[1].trim();
+		const updatedTokensList = tokensList ? `${tokensList}, ${testnetToken}` : testnetToken;
+		content = content.replace(
+			sepoliaRegex,
+			`const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token[] = [${updatedTokensList}];`
+		);
+	}
+
+	const importStatement = `import { ${mainnetToken}, ${testnetToken} } from '$env/${newFileName.replace('.ts', '')}';\n`;
+	content = importStatement + content;
+
+	await fs.writeFile(filePath, content);
+	console.log(`Updated ${filePath}`);
+};
+
+const fetchSupportedTokens = async (url) => {
+	try {
+		const html = await fetchHtml(url);
+		const $ = load(html);
+
+		const tokens = [];
+		$('#supported-ckerc20-tokens + table tbody tr').each((index, element) => {
+			const symbol = $(element).find('td').eq(0).text().trim();
+			const contractAddress = $(element).find('td').eq(2).text().trim();
+			tokens.push({ symbol, contractAddress });
+		});
+
+		console.log(
+			`Found ${tokens.length} tokens on ${url}: ${tokens.map((t) => t.symbol).join(', ')}`
+		);
+
+		return tokens;
+	} catch (err) {
+		console.error(`Error fetching supported tokens on ${url}:\n${err}`);
+		process.exit(1);
+	}
+};
+
+const main = async () => {
+	const prodTokens = await fetchSupportedTokens(PROD_DASHBOARD_URL);
+	const testnetTokens = await fetchSupportedTokens(TESTNET_DASHBOARD_URL);
+
+	const testnetTokenMap = new Map(
+		testnetTokens.map((token) => [token.symbol.replace('ckSepolia', 'ck'), token.contractAddress])
+	);
+
+	const tokensToProcess = [];
+
+	for (const token of prodTokens) {
+		const testnetContractAddress = testnetTokenMap.get(token.symbol);
+		if (!testnetContractAddress) {
+			continue;
+		}
+
+		const tokenName = token.symbol.slice(2);
+		const tokenFilePath = path.join(SRC_PATH, `tokens.${tokenName.toLowerCase()}.env.ts`);
+		try {
+			await fs.access(tokenFilePath);
+		} catch (err) {
+			tokensToProcess.push({
```

Comment:
When I use `access` this is how I check if the error is not found or another error:

```
export const junoConfigExist = async (params: {filename: ConfigFilename}): Promise<boolean> => {
  try {
    const {configPath} = junoConfigFile(params);
    await access(configPath);
    return true;
  } catch (err: unknown) {
    if (err instanceof Error && 'code' in err && (err as NodeJS.ErrnoException).code === 'ENOENT') {
      return false;
    } else {
      throw err;
    }
  }
};
```

sharing just in case in intersting for this use case.

Reply:
yes, i will use, better